### PR TITLE
MAINT: Update `array-api-tests` job

### PR DESCRIPTION
**Pull request by mtsokol**
Follow-up for #25167

This PR bumps `array-api-tests` to the latest version and cleans up `tools/ci/array-api-skips.txt` file (e.g. `fft` test suite is fixed now). 

Here's a discussion issue for `np.nonzero` 0D input test: https://github.com/numpy/numpy/issues/26238

### CODE REVIEW

**Comment by seberg** [comment]
Ah, hadn't noticed this and pinged @asmeurer on another PR, but maybe you can just include the fix here?  The tests are failing with:
```
=========================== short test summary info ============================
FAILED array_api_tests/test_signatures.py::test_func_signature[vecdot] - AssertionError: Argument 'axis' missing from signature
assert 'axis' in odict_keys(['args', 'kwargs'])
 +  where 'axis' = <Parameter "axis: int = -1">.name
 +  and   odict_keys(['args', 'kwargs']) = <built-in method keys of mappingproxy object at 0x7f70da8082e0>()
 +    where <built-in method keys of mappingproxy object at 0x7f70da8082e0> = mappingproxy(OrderedDict([('args', <Parameter "*args">), ('kwargs', <Parameter "**kwargs">)])).keys
 +      where mappingproxy(OrderedDict([('args', <Parameter "*args">), ('kwargs', <Parameter "**kwargs">)])) = <Signature (*args, **kwargs)>.parameters
=========== 1 failed, 900 passed, 30 skipped, 119 warnings in 5.19s ============
```
which is just an overzealous test because inspection is failing.  Maybe we can try to fix inspection, skip the test (dunno how easy it is to implement inspection there).

**Comment by mtsokol** [comment]
I've just been working on it, it's surprising:
```python
import numpy as np
import inspect

inspect.signature(np.vecdot)
```
locally gives `ValueError: callable <ufunc 'vecdot'> is not supported by signature` which is correct behavior.

The `array-api-tests` tests it in `_test_func_signature`:

```python
try:
    sig = signature(func)
except ValueError:
    try:
        _test_uninspectable_func(stub.__name__, func, stub_sig)
    except Exception as e:
        raise e from None  # suppress parent exception for cleaner pytest output
else:
    _test_inspectable_func(sig, stub_sig)
```
Where locally it raises in `try` and calls `_test_uninspectable_func` as expected.

But in the CI `signature(np.vecdot)` returns `Signatue(*arg, **kwarg)` (!?)

**Comment by mtsokol** [comment]
For now I skipped this test with a `TODO`

File: tools/ci/array-api-skips.txt [hunk 1]
```diff
@@ -1,50 +1,22 @@
...
 array_api_tests/test_signatures.py::test_func_signature[sort]
 
-# assertionError: out.dtype=float32, but should be float64 [sum(float32)]
-array_api_tests/test_statistical_functions.py::test_sum
+# nonzero for 0D should error
+array_api_tests/test_searching_functions.py::test_nonzero_zerodim_error
+
+# TODO: check why in CI `inspect.signature(np.vecdot)` returns (*arg, **kwarg)
+#       instead of raising ValueError. mtsokol: couldn't reproduce locally
+array_api_tests/test_signatures.py::test_func_signature[vecdot]
```
Comment (charris) [inline tools/ci/array-api-skips.txt:22]: This is something that isn't repeatable in CI, or at least rerunning succeeded for me.

**Comment by charris** [comment]
Thanks Mateusz.

**Comment by asmeurer** [comment]
Ping @honno on the array-api-tests failure.

**Comment by asmeurer** [comment]
Also an aside, but is there a reason ufuncs don't define `__signature__` so that they do work with `inspect.signature`?

**Comment by mattip** [comment]
No real reason. There might be a general issue with c-extensions and `__signature__`, progress on thinking about this is welcome.

**Comment by asmeurer** [comment]
I found https://github.com/numpy/numpy/issues/8734.

**Pull request merged**